src/providers/SubscriptionProvider/SubscriptionProvider.actions.js

```diff
@@ -43,6 +44,8 @@ export function updateIsOnTrialPeriod() {
         isOnTrialPeriod
       })
     );
+
+    if (!isOnTrialPeriod) disablePremiumFeaturesOnTrialPeriodEnded();
```

Comment:
I think this will also disable the feature in users who are premium

Reply:
Really on truth. Good Review!